protos/follow_me/follow_me.proto

```diff
@@ -31,18 +31,23 @@ service FollowMeService {
 // Configuration type.
 message Config {
     // Direction relative to the target that the vehicle should follow
+    // Measured in reference to target's course (where it's moving towards)
     enum FollowDirection {
-        FOLLOW_DIRECTION_NONE = 0; // Do not follow
-        FOLLOW_DIRECTION_BEHIND = 1; // Follow from behind
-        FOLLOW_DIRECTION_FRONT = 2; // Follow from front
-        FOLLOW_DIRECTION_FRONT_RIGHT = 3; // Follow from front right
-        FOLLOW_DIRECTION_FRONT_LEFT = 4; // Follow from front left
+        FOLLOW_DIRECTION_BEHIND = 0; // Follow from behind
+        FOLLOW_DIRECTION_FRONT = 1; // Follow from front
+        FOLLOW_DIRECTION_FRONT_RIGHT = 2; // Follow from front right
+        FOLLOW_DIRECTION_FRONT_LEFT = 3; // Follow from front left
+        FOLLOW_DIRECTION_RIGHT = 4; // Follow from right
+        FOLLOW_DIRECTION_LEFT = 5; // Follow from left
+        FOLLOW_DIRECTION_BEHIND_RIGHT = 6; // Follow from behind right
+        FOLLOW_DIRECTION_BEHIND_LEFT = 7; // Follow from behind left
+        FOLLOW_DIRECTION_NONE = 8; // Follow direction invalid (defaults to Behind)
```

Comment:
Changing the indices will have weird protobuf consequences :innocent:. Again those indices are exclusively reserved to the proto format, and don't compare to an enum value in a language like C++.

```suggestion
        FOLLOW_DIRECTION_NONE = 0; // Follow direction invalid (defaults to Behind)
        FOLLOW_DIRECTION_BEHIND = 1; // Follow from behind
        FOLLOW_DIRECTION_FRONT = 2; // Follow from front
        FOLLOW_DIRECTION_FRONT_RIGHT = 3; // Follow from front right
        FOLLOW_DIRECTION_FRONT_LEFT = 4; // Follow from front left
        FOLLOW_DIRECTION_RIGHT = 5; // Follow from right
        FOLLOW_DIRECTION_LEFT = 6; // Follow from left
        FOLLOW_DIRECTION_BEHIND_RIGHT = 7; // Follow from behind right
        FOLLOW_DIRECTION_BEHIND_LEFT = 8; // Follow from behind left
```

Reply:
Alright! But how do we deal with the enum values not matching the expected values from the Autopilot then?

https://github.com/mavlink/MAVSDK/blob/main/src/mavsdk/plugins/follow_me/follow_me_impl.cpp#L99

For example here, the enum class gets directly casted into integer & gets sent to the Autopilot. How can I adapt current changes to this code then? Doing Switch case and sending different enum values? (since they will have different values assigned for each follow direction :thinking: )

Reply:
Ohhhh so technically all the code in MAVSDK that has those enums must do a switch case to manually map out the actual values! Then this follow me code was initially dangerous :smirk: